js/viz.js

```diff
@@ -98,38 +129,39 @@ const viz = {
     label.attr('class', `textLabel text${index} invisible`);
   },
 
-  drawLabels(nodes) {
-    this.allLabels = this.allLabels.data(nodes, (d) => d.hostname);
-    this.allLabels.exit().remove();
-
-    let newText = this.allLabels.enter();
-    newText = newText.append('text');
-    newText.attr('class', (d, i) => `textLabel text${i}`);
-    newText.text((d) => d.hostname);
-    newText.attr('fill', 'white');
-
-    this.allLabels = newText.merge(this.allLabels);
+  drawLinks() {
+    this.context.beginPath();
+    this.lines.each((d) => {
+      this.context.moveTo(d.source.x, d.source.y);
+      this.context.lineTo(d.target.x, d.target.y);
+    });
+    this.context.closePath();
+    this.context.strokeStyle = '#ccc';
+    this.context.stroke();
   },
 
-  drawLinks(links) {
-    this.allLines = this.allLines
-      .data(links, (d) => `${d.source.hostname}-${d.target.hostname}`);
+  virtualDom(type, elements) {
```

Comment:
This isn't virtual, if we are keeping with this method then rename this to be about being fake dom perhaps.